### ISSUE TITLE
/etc/cron.daily/netdata-updater -f $'\E(B\E[0m\\n' /dev/fd/63: line 86: : command not found

### ISSUE THREAD

**Issue by slrslr** [netdata/netdata#5570]
Hello, i had netdata installed but i uninstalled it ([using commands here](https://github.com/netdata/netdata/issues/103#issuecomment-204076030)).
Later, today i tried to install it:

bash <(curl -Ss https://my-netdata.io/kickstart.sh)
 --- Re-installing netdata... --- 
$'[\E[2m/home/lynx\E(B\E[0m]# \E[1m\E[33m' '' /etc/cron.daily/netdata-updater -f $'\E(B\E[0m\\n' /dev/fd/63: line 86: : command not found
 FAILED   /etc/cron.daily/netdata-updater -f 

 ABORTED  Failed to forcefully update netdata

---
this is Debian based Linux "4.20.0-3.1-liquorix-amd64 #1 ZEN SMP PREEMPT liquorix 4.20-1.1~sid (2019-01-20) x86_64 GNU/Linux"

**Comment by cakrit** [comment]
That thread is two years old. Please use this method to uninstall: https://docs.netdata.cloud/packaging/installer/uninstall/

**Comment by slrslr** [comment]
That worked. Thank you. It installed without problem now.

**Comment by MichaelRiss** [comment]
I also stumbled over this problem but coming from a slightly different angle.
We install netdata with ansible and there I just call 
```
bash <(curl -Ss https://my-netdata.io/kickstart.sh) --dont-wait
```
in a role to install it, to update it and to ensure it is installed.
The `kickstart.sh` script handles the case of reinstalling netdata by handing over the task to the command `${sudo}" "${updater} -f` and tries to execute this command within quotation marks.
On my setup this triggers a couple of bugs:
- (minor) as far as I can see the `-f` option has no effect on `/etc/cron.daily/netdata-updater`, but I'm not sure about the other installer `/etc/periodic/daily/netdata-updater` - but my feeling is that the script is the same there. If I'm right, this `-f` can be dropped.
- I run my ansible role as root, which makes `kickstart.sh` assign `sudo=""`. This way the command string starts with a space (the space between `${sudo}` and `${updater}`).
- Finally the command is executed in the `run()` function as `"${@}"` (with quotation marks!) which in my case tries to execute the executable `" /etc/cron.daily/netdata-updater -f"`, which does not exist in the file system.

I see that two months ago the `run()` function with the `"${@}"` invocation in the `kickstart.sh` has been overhauled, so I'm not sure that this problem is really the same as the one from two years ago @cakrit  mentioned. To me it seems it's something that has been introduced with https://github.com/netdata/netdata/commit/b81bba75f1fc6f275d0acb4a3fc68a7b8c062c65.

Solution ideas (coming from a guy looking at this code for barely 2h, so please double check):
- most effective: replace `"${@}"` with `${@}` for execution in `run()`
- alternative:
  - drop the `-f` option
  - eliminate the leading space in the command when running the updater as root with e.g.
    ```
    command="${sudo} ${updater}"
    command="$(echo -e "${command}" | awk '{$1=$1};1')"
    run "${command}" || fatal "Failed to forcefully update netdata"
    ```
    in the section below `progress "Re-installing netdata..."`.

**Comment by cakrit** [comment]
Excellent report @MichaelRiss, thanks! Can you please copy paste this exact post on a new issue (type bug)? We'll get on the fix, but it's really a completely different issue, so we shouldn't continue here.

**Comment by MichaelRiss** [comment]
Thanks for the positive feedback, I opened a new issue https://github.com/netdata/netdata/issues/5584 for it.

**Comment by paulkatsoulakis** [comment]
This was fixed with PR #5581 and will be included in the next release.
Sorry for the inconvenience, it should have been included in v1.13.0

Please do open a new issue if after the next available release things continue to break
